docs/DeveloperGuide.adoc

```diff
@@ -730,92 +730,114 @@ image::clone/cloneDG/CloneActivityDiagram.png[]
+Step 3. The user executes *undo* to remove Laksa from `**TransactionList**`.
 The `**UndoCommand**` invokes the method `Model#getPreviousUndoableCommand()` to retrieve the `**AddExpenseCommand**` object.
 `**AddExpenseCommand**` is popped from `UndoStack` and pushed into `RedoStack` before being returned to the callee.
 
-image::UndoAndRedo/undo2.PNG[326, 300]
+image::UndoAndRedo/undo.PNG[, 250]
 
-The sequence diagram shows how *undo* work for *add_expense* command:
+The sequence diagram shows how *undo* works for `**AddExpenseCommand**`:
 
-.Sequence diagram of how *undo* command is applied at the back-end
+.Sequence diagram of how `**UndoCommand**` is applied at the back-end
 image::UndoAndRedo/UndoAddExpenseSequenceDiagram.png[]
+// end::undo[]
+
+[[GeneralUndoSequence]]
+A general sequence diagram to illustrate how *undo* works for undoable commands:
+
+.General sequence diagram of how `**UndoCommand**` is applied at the back-end for undoable commands
+image::UndoAndRedo/UndoGeneralSequenceDiagram.png[]
 
-Step 6. The user executes *redo* to add Laksa back to `**TransactionList**`.
+// tag::redo[]
+Step 4. The user executes *redo* to add Laksa back to `**TransactionList**`.
 The `**RedoCommand**` invokes the method `Model#getUndoneCommand()` to retrieve the `**AddExpenseCommand**` object.
 `**AddExpenseCommand**` is popped from `RedoStack` and pushed into `UndoStack` before being returned to the callee.
 
-image::UndoAndRedo/redo.PNG[325, 300]
+image::UndoAndRedo/redo.PNG[, 250]
 
-NOTE: All commands stored in `RedoStack` are purged when the user executes another undoable command.
+NOTE: All undoable commands objects stored in `RedoStack` are purged when the user executes another undoable command.
 
-The sequence diagram shows how *redo* work for *add_expense* command:
+The sequence diagram shows how *redo* work for `**AddExpenseCommand**`:
 
-.Sequence diagram of how *redo* command is applied at the back-end
+.Sequence diagram of how `**RedoCommand**` is applied at the back-end
 image::UndoAndRedo/RedoAddExpenseSequenceDiagram.png[]
+// end::redo[]
 
+[[GeneralRedoSequence]]
+A general sequence diagram to illustrate how *redo* work for undoable commands:
+
+.General sequence diagram of how `**RedoCommand**` is applied at the back-end for undoable commands
+image::UndoAndRedo/RedoGeneralSequenceDiagram.png[]
+
+// tag::designconsideration[]
 The following activity diagram shows what happens if the user executes a command:
 
 .Activity diagram of the command execution.
 image::UndoAndRedo/RunCommandActivityDiagrams.png[]
 
 ==== Design considerations
 
+This section shows some of the design considerations taken before implementing the undo and redo features.
+
+===== Aspect: Algorithm used to implements undo and redo features
+
+* Alternative 1 (current choice): Provides different sets of implementation for each command so that the command knows how to undo and redo itself.
```

Comment:
How about:
_Provide_ different _implementations_ for each command ...

Reply:
Fixed